ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,611 @@
+
+		networkSecurityGroup, err := nsgClient.AddRule(nsgID, &networkSecurityGroupAddRule)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		_, err = isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx, nsgClient, nsgID, *networkSecurityGroup.ID, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.SetId(fmt.Sprintf("%s/%s/%s", cloudInstanceID, nsgID, *networkSecurityGroup.ID))
+	}
+
+	return resourceIBMPINetworkSecurityGroupRuleRead(ctx, d, meta)
+}
+
+func resourceIBMPINetworkSecurityGroupRuleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, nsgID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	nsgClient := instance.NewIBMIPINetworkSecurityGroupClient(ctx, sess, cloudInstanceID)
+	networkSecurityGroup, err := nsgClient.Get(nsgID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_CRN, networkSecurityGroup.Crn)
+
+	if networkSecurityGroup.Members != nil {
+		members := []map[string]interface{}{}
+		for _, mbr := range networkSecurityGroup.Members {
+			mbrMap := networkSecurityGroupMemberToMap(mbr)
+			members = append(members, mbrMap)
+		}
+		d.Set(Attr_Members, members)
+	}
+	d.Set(Arg_Name, networkSecurityGroup.Name)
+	d.Set(Attr_NetworkSecurityGroupID, networkSecurityGroup.ID)
+	if networkSecurityGroup.Rules != nil {
+		rules := []map[string]interface{}{}
+		for _, rule := range networkSecurityGroup.Rules {
+			ruleMap := networkSecurityGroupRuleToMap(rule)
+			rules = append(rules, ruleMap)
+		}
+		d.Set(Attr_Rules, rules)
+	}
+
+	return nil
+}
+
+func resourceIBMPINetworkSecurityGroupRuleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId("")
+	return nil
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, memberID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Available},
+		Refresh:    isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client, id, memberID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
+	}
+
+	return stateConf.WaitForStateContext(ctx)
+}
+
+func isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client *instance.IBMPINetworkSecurityGroupClient, id, memberID string) retry.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		networkSecurityGroup, err := client.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if networkSecurityGroup.Members != nil {
+			for _, mbr := range networkSecurityGroup.Members {
+				if *mbr.ID == memberID {
+					return networkSecurityGroup, State_Available, nil
+				}
+
+			}
+		}
+		return networkSecurityGroup, State_Pending, nil
+	}
+}
```

Comment:
Why are you checking for member if this is after adding a rule?

Reply:
I copy pasted and forgot to replace them, should be checking for the new rule id.